### DISCUSSION TITLE
How to set the unique constraints on edge

### DISCUSSION THREAD

**Discussion by vitaliishandra** [JanusGraph/janusgraph#3337] [Q&A] [answered]
I have a query that creates the edge with label 'License' between two vertexes 'User1' and 'Content1' if it does not exist. We have few instances of our service and this query can be executed in paralell in scope of different operations. As a result we will receive few same 'License' edges between 'User1' and 'Content1'.
We need to gurantee that only one 'License' edge should be creted between same vertexes.

As we understand to do that we need to change label multiplicity from default 'MULTI' to 'SINGLE'. If so, we need to understand the impact on existing queries and data:
1) If query will try to create the second 'License' edge between the two same vertexes will it fail or just update the properties of existing edge?
2) Can we change the multiplicity of existing Edge using the following command mgmt.makeEdgeLabel('License').multiplicity(SINGLE).make() and not impact existing data?

**Comment by vtslab** [comment, marked as answer]
Please check:
https://docs.janusgraph.org/advanced-topics/eventual-consistency/   in particular ConsistencyModifier.LOCK
https://docs.janusgraph.org/advanced-topics/eventual-consistency/#data-consistency-without-locks

Regarding your questions:
1. many situations possible regarding the type and number of nodes of the storage backend and the exact schema definition
2. no, schema element definitions cannot be changed once committed to the graph, see:
https://lists.lfaidata.foundation/g/janusgraph-users/topic/79936756#3847
